Fundamentals block - main and optional tasks

lerik14 opened the pull request.

lerik14 committed:
Add main and optional tasks for Java fundamentals block

Orkyboy commented on java_fundamentals_block/main_task/src/main/java/Actions.java:
```diff
@@ -0,0 +1,112 @@
+            }
+        }
+        System.out.println("The sum of all numbers is " + sum);
+        System.out.println("The product of all numbers is " + multiple);
```
product? is it typo?

lerik14 replied:
No. The product of numbers is the result you get when you multiply them together.

Orkyboy commented on java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java:
```diff
@@ -0,0 +1,82 @@
+        System.out.println("The length of this element is " + elemLength(longElem));
+    }
+
+    public static void sort(int[] array) {
```
rename to sortByLength to emphasize what function does

lerik14 replied:
done

Orkyboy commented on java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java:
```diff
@@ -0,0 +1,82 @@
+
+    private static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public static int[] inputElem() throws IOException {
```
i suggest renaming to getElementsArrayFromConsole

lerik14 replied:
good idea. Done

Orkyboy commented on java_fundamentals_block/optional_task/src/main/java/OptionalTask1.java:
```diff
@@ -0,0 +1,82 @@
+        return array;
+    }
+
+    public static void findShortElement(int[] array) {
```
I think in current implementation these functions do 2 things:
1. action (like find shortest element here)
2. printing to console

I suggest leave only single responsibility to the function and move printing out to `OptionalTaskMain.java`

so converted this function will be like
```    
public static int findShortestElement(int[] array) {
        int shortElem = array[0];
        for (int i = 1; i < array.length; i++) {
            if (array[i] < shortElem) {
                shortElem = array[i];
            }
        }
   return shortElem;
}
```

And then in OptionalTaskMain
```
int shortestElem = OptionalTask1.findShortElement(array);
        System.out.println("The longest element is: " + shortestElem);
        System.out.println("The length of this element is: " + OptionalTask1.elemLength(shortestElem));
```

lerik14 replied:
Done for this and other methods

lerik14 committed:
Add code review fixed
Changed implementation findShortestElement, findLongestElement and elementsLongerThanAverageLength methods - there is no printing to console.

lerik14 committed:
Changed averageLength

lerik14 committed:
Changed averageLength type
